src/eval_utils/run.ts

```diff
@@ -0,0 +1,725 @@
+        `${CYAN}${type.charAt(0).toUpperCase() + type.slice(1)} Version ID: ${
+            hlFile.versionId
+        }${RESET}`,
+    );
+    console.log(`${CYAN}Run ID: ${runId}${RESET}`);
+
+    // Generate locally if a function is provided
+    if (function_) {
+        console.log(
+            `${CYAN}\nRunning ${hlFile.name} over the Dataset ${hlDataset.name}${RESET}`,
+        );
+        const totalDatapoints = hlDataset.datapoints!.length;
+        progressBar.start(totalDatapoints, 0);
+        await pMap(
+            hlDataset.datapoints!,
+            async (datapoint) => {
+                await processDatapoint(datapoint, runId);
+            },
+            { concurrency: workers },
+        );
+        progressBar.stop();
+    } else {
+        // TODO: trigger run when updated API is available
+        console.log(
+            `${CYAN}\nRunning ${hlFile.name} over the Dataset ${hlDataset.name}${RESET}`,
+        );
+    }
+
+    // Wait for the Evaluation to complete then print the results
+    let stats;
+    do {
+        stats = await client.evaluations.getStats(evaluation.id);
+        console.log(`\r${stats.progress}`);
+        if (stats.status !== "completed") {
+            await new Promise((resolve) => setTimeout(resolve, 5000));
+        }
+    } while (stats.status !== "completed");
+    console.log(stats.report);
+
+    const checks: EvaluatorCheck[] = [];
+    if (
+        evaluators.some((evaluator) => evaluator.threshold !== undefined) ||
+        stats.runStats.length > 1
+    ) {
+        for (const evaluator of evaluators) {
+            const [_, score, delta] = checkEvaluationImprovement(
+                evaluation,
+                evaluator.path!,
+                stats,
+                runId,
+            );
+            let thresholdCheck = undefined;
+            if (evaluator.threshold !== undefined) {
+                thresholdCheck = score >= evaluator.threshold;
+                thresholdCheck = checkEvaluationThreshold(
+                    evaluation,
+                    stats,
+                    evaluator.path!,
+                    evaluator.threshold,
+                    runId,
+                );
+            }
+            checks.push({
+                path: evaluator.path!,
+                // TODO: Add back in with number valence on Evaluators
+                // improvementCheck: improvementCheck,
+                score: score,
+                delta: delta,
+                threshold: evaluator.threshold,
+                thresholdCheck: thresholdCheck,
+                evaluationId: evaluation.id,
+            });
+        }
+    }
+
+    console.info(`\n${CYAN}View your Evaluation:${RESET}\n${evaluation.url}\n`);
+    return checks;
+}
+
+function getLogFunction(
+    client: HumanloopClient,
+    type: FileType,
+    fileId: string,
+    versionId: string,
+    runId: string,
+) {
+    /** Returns the appropriate log function pre-filled with common parameters. */
```

Comment:
docstring should be above function right?

Reply:
ack changed